src/dawn/Optimizer/PassStageSplitter.cpp

```diff
@@ -57,6 +70,7 @@ bool PassStageSplitter::run(
         std::shared_ptr<iir::DependencyGraphAccesses> newGraph, oldGraph;
         newGraph =
             std::make_shared<iir::DependencyGraphAccesses>(stencilInstantiation->getMetaData());
+        oldGraph = newGraph->clone();
```

Comment:
this changes what we're doing. I still think what should happen is that lines 71 to 73 should be removed and we should be in the clear. This is since these are just a copy/paste error and this is why we're getting the nullptr exception since 95-97 is overwriting what you dump in 71-73.

Reply:
The nullptr exception happens when a conflict is detected on the first iteration of this loop: https://github.com/MeteoSwiss-APN/dawn/blob/cc0613e43a3b6a77fee7de783c04ca113c407e70/src/dawn/Optimizer/PassStageSplitter.cpp#L76-L83
Even if the dumping code is removed, still a null pointer would be saved in the `graphs` container:
https://github.com/MeteoSwiss-APN/dawn/blob/cc0613e43a3b6a77fee7de783c04ca113c407e70/src/dawn/Optimizer/PassStageSplitter.cpp#L100-L106
`oldGraph` must be initialized somehow before being used and i thought that having it initialized as a clone of `newGraph` would keep the same semantics for the first iteration as for another iteration, as each iteration ends this way:
https://github.com/MeteoSwiss-APN/dawn/blob/cc0613e43a3b6a77fee7de783c04ca113c407e70/src/dawn/Optimizer/PassStageSplitter.cpp#L121

Reply:
sounds reasonable!
Are we still removing the reduntand write to dot though?

Reply:
Would it be ok to do it in this PR? Or opening a new one would be a better choice?

Reply:
you're already touching this file heavily, I'd do it here

Reply:
If I understand correctly what you propose is to remove the `oldGraph->toDot(...)` call inside the loop and only have `newGraph->toDot(...)` outside. However, in this way, we would only dump the dependency graph of the first stage (from the last split). Isn't the intended functionality to dump the dep. graphs of every stage generated by this pass?

Reply:
yeah, seems like you're correct. I misread one line. We should definitely keep it